wire-ios/Wire-iOS/Sources/UserInterface/Conversation Options/ConversationGuestOptionsViewController.swift

```diff
@@ -57,11 +61,30 @@ final class ConversationGuestOptionsViewController: UIViewController,
         viewModel.delegate = self
     }
 
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
+
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(handleGuestLinkNotification(_ :)),
+            name: .didCreateSecureGuestLink,
+            object: nil
+        )
```

Comment:
Can we use the block based API to avoid the `objc` annotation and selectors? That one returns a observer token `any NSObjectProtocol`.

Reply:
Done!